huaweicloud/services/modelarts/resource_huaweicloud_modelarts_network.go

```diff
@@ -0,0 +1,540 @@
+			}
+
+			createNetworkWaitingPath := createNetworkWaitingClient.Endpoint + createNetworkWaitingHttpUrl
+			createNetworkWaitingPath = strings.ReplaceAll(createNetworkWaitingPath, "{project_id}", createNetworkWaitingClient.ProjectID)
+			createNetworkWaitingPath = strings.ReplaceAll(createNetworkWaitingPath, "{id}", d.Id())
+
+			createNetworkWaitingOpt := golangsdk.RequestOpts{
+				KeepResponseBody: true,
+				OkCodes: []int{
+					200,
+				},
+				MoreHeaders: map[string]string{"Content-Type": "application/json"},
+			}
+
+			createNetworkWaitingResp, err := createNetworkWaitingClient.Request("GET", createNetworkWaitingPath, &createNetworkWaitingOpt)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+
+			createNetworkWaitingRespBody, err := utils.FlattenResponse(createNetworkWaitingResp)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+
+			statusRaw := utils.PathSearch(`status.phase`, createNetworkWaitingRespBody, "")
+
+			status := fmt.Sprintf("%v", statusRaw)
+
+			targetStatus := []string{
+				"Active",
+			}
+			if utils.StrSliceContains(targetStatus, status) {
+				return createNetworkWaitingRespBody, "COMPLETED", nil
+			}
+
+			pendingStatus := []string{
+				"Creating", "",
+			}
+			if utils.StrSliceContains(pendingStatus, status) {
+				return createNetworkWaitingRespBody, "PENDING", nil
+			}
+
+			return createNetworkWaitingRespBody, status, nil
+		},
+		Timeout:      t,
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func resourceModelartsNetworkRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getModelartsNetwork: Query the Modelarts network.
+	var (
+		getModelartsNetworkHttpUrl = "v1/{project_id}/networks/{id}"
+		getModelartsNetworkProduct = "modelarts"
+	)
+	getModelartsNetworkClient, err := cfg.NewServiceClient(getModelartsNetworkProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating ModelArts client: %s", err)
+	}
+
+	getModelartsNetworkPath := getModelartsNetworkClient.Endpoint + getModelartsNetworkHttpUrl
+	getModelartsNetworkPath = strings.ReplaceAll(getModelartsNetworkPath, "{project_id}", getModelartsNetworkClient.ProjectID)
+	getModelartsNetworkPath = strings.ReplaceAll(getModelartsNetworkPath, "{id}", d.Id())
+
+	getModelartsNetworkOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+
+	getModelartsNetworkResp, err := getModelartsNetworkClient.Request("GET", getModelartsNetworkPath, &getModelartsNetworkOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Modelarts network")
+	}
+
+	getModelartsNetworkRespBody, err := utils.FlattenResponse(getModelartsNetworkResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
how about extracting these codes into a separate function, the other three waiting functions all use these codes

Reply:
OK

Reply:
ok